pom.xml

```diff
@@ -10,13 +10,45 @@
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     </properties>
-
+    <packaging>jar</packaging>
     <dependencies>
 
+        <dependency>
+            <groupId>com.sparkjava</groupId>
+            <artifactId>spark-core</artifactId>
+            <version>2.8.0</version>
+            <exclusions>
+                <exclusion>
+                    <!-- this pom will declare the version of sparkjava -->
+                    <groupId>org.slf4j</groupId>
+                    <artifactId>spark-core</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+
+
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <version>8.0.18</version>
+        </dependency>
+        <dependency>
+            <groupId>com.google.cloud.sql</groupId>
+            <artifactId>mysql-socket-factory-connector-j-8</artifactId>
+            <version>1.0.15</version>
+        </dependency>
+
```

Comment:
You probably dealt with it, but can't see the DB url so will mention it. I asked below about updating JDBC URL. Using their socket factory is one of the options in the URL. I believe same effect can be obtained when creating connection in code.

Reply:
Yep, I used the cloudsqlproxy-style URLs.  You can check  the secrets [here](https://console.cloud.google.com/security/secret-manager?folder=&organizationId=&project=broad-ddp-dev).   in  a UI!  With secret versions!  And no weird "don't forget `.data` and `jq` stuff"...

Reply:
I followed GAE's recommendation and setup the URL like below.
Requires granting permissions. If the proxy thing works then maybe its all good. But there might or might not be a performance penalty.
jdbc:mysql:///pepperdev?cloudSqlInstance=broad-ddp-dev:us-central1:ddp-dev&socketFactory=com.google.cloud.sql.mysql.SocketFactory&user=XXXX&password=YYYYY@&characterEncoding=UTF-8&useLegacyDatetimeCode=false&serverTimezone=UTC&useSSL=false&sessionVariables=innodb_strict_mode=on,tx_isolation='READ-COMMITTED',sql_mode='TRADITIONAL'

Reply:
Gaah, I totally forgot the `UTC` url params

Reply:
I don't think you need sqlproxy to run in GAE. Just need to grant permissions to the service. Read somewhere (don't have the reference, sorry) that sqlproxy introduced latency.

Reply:
Sorry for the confusion--when I said "cloudsqlproxy-style", I meant the urls like `cloudSqlInstance=broad-ddp-dev:us-central1:ddp-dev`.  So not the cloudsqlproxy per se, but urls that reference gcp project, zone, and cloudsql instance name instead of IP.